geopandas/array.py

```diff
@@ -215,6 +230,8 @@ class GeometryArray(ExtensionArray):
 
     def __init__(self, data, crs=None):
         if isinstance(data, self.__class__):
+            if not crs:
+                crs = data.crs
```

Comment:
Is there a test that asserts this behaviour? (prioritizing the passed crs over the data's crs)

Reply:
No, I will add one. This covers `GeometryArray(geom_array)` pattern which I did not think about before.